### ISSUE TITLE
missing DeepCopyObject method

### ISSUE THREAD

**Issue by dchittibala** [xcoulon/podset-operator#5]
Following through the documentation and go through the build stage, I get the following error
```➜ operator-sdk build podset-operator
# github.com/podset-operator/pkg/controller/podset
pkg/controller/podset/podset_controller.go:113:4: cannot use &client.ListOptions literal (type *client.ListOptions) as type runtime.Object in argument to r.client.List:
	*client.ListOptions does not implement runtime.Object (missing DeepCopyObject method)
pkg/controller/podset/podset_controller.go:113:4: cannot use existingPods (type *"k8s.io/api/core/v1".PodList) as type client.ListOption in argument to r.client.List:
	*"k8s.io/api/core/v1".PodList does not implement client.ListOption (missing ApplyToList method)
Error: failed to build operator binary: (failed to exec []string{"go", "build", "-o", "/Users/a6002031/go/src/github.com/podset-operator/build/_output/bin/podset-operator", "-gcflags", "all=-trimpath=/Users/a6002031/go/src/github.com", "-asmflags", "all=-trimpath=/Users/a6002031/go/src/github.com", "github.com/podset-operator/cmd/manager"}: exit status 2)
Usage:
  operator-sdk build <image> [flags]

Flags:
      --go-build-args string      Extra Go build arguments as one string such as "-ldflags -X=main.xyz=abc"
  -h, --help                      help for build
      --image-build-args string   Extra image build arguments as one string such as "--build-arg https_proxy=$https_proxy"
      --image-builder string      Tool to build OCI images. One of: [docker, podman, buildah] (default "docker")

Global Flags:
      --verbose   Enable verbose logging
```

Operator-sdk Version
```➜ operator-sdk version
operator-sdk version: "v0.12.0", commit: "2445fcda834ca4b7cf0d6c38fba6317fb219b469", go version: "go1.13.4 darwin/amd64"
```

It seems like I get the error on the podset_controller.go at line 110
```	err = r.client.List(context.TODO(),
		&client.ListOptions{
			Namespace:     request.Namespace,
			LabelSelector: labels.SelectorFromSet(lbls),
		},
		existingPods)
```

Any help is appreciated.

**Comment by xcoulon** [comment]
hello @reddydinesh427 

thanks for reporting this issue. I just upgraded the project to Go 1.13 with modules and Kubernetes 1.15 dependencies (see https://github.com/xcoulon/podset-operator/blob/master/go.mod) and I was able to reproduce the problem that you reported. I'm going to look into that.

**Comment by xcoulon** [comment]
@reddydinesh427 that should be fixed in 06c6758. The `client.List()` method signature changed and the parameters needed to be switched :/

Could you confirm that it works for you now, please?

**Comment by dchittibala** [comment]
The fix works. Thanks
